contrib/spring/src/main/java/io/opencensus/contrib/spring/autoconfig/TraceWebAsyncClientAutoConfiguration.java

```diff
@@ -0,0 +1,85 @@
+ */
+
+package io.opencensus.contrib.spring.autoconfig;
+
+import io.opencensus.common.ExperimentalApi;
+import io.opencensus.contrib.spring.instrument.web.client.TracingAsyncClientHttpRequestInterceptor;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.List;
+import javax.annotation.PostConstruct;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.autoconfigure.AutoConfigureAfter;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnBean;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnClass;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnProperty;
+import org.springframework.boot.context.properties.EnableConfigurationProperties;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.ComponentScan;
+import org.springframework.context.annotation.Configuration;
+
+/**
+ * {@link org.springframework.boot.autoconfigure.EnableAutoConfiguration Auto-configuration} enables
+ * span information propagation for {@link org.springframework.web.client.AsyncRestTemplate}.
+ *
+ * @since 0.23.0
+ */
+@Configuration
+@ComponentScan(basePackages = "io.opencensus")
+@ConditionalOnProperty(value = "opencensus.spring.enabled", matchIfMissing = true)
+@ConditionalOnClass(org.springframework.web.client.AsyncRestTemplate.class)
+@EnableConfigurationProperties(OpenCensusProperties.class)
+@AutoConfigureAfter(OpenCensusAutoConfiguration.class)
+@ExperimentalApi
+@SuppressWarnings("deprecation")
+public class TraceWebAsyncClientAutoConfiguration {
+  @Configuration
+  @ConditionalOnBean(org.springframework.web.client.AsyncRestTemplate.class)
+  static class AsyncRestTemplateCfg {
+
+    @Bean
+    public TracingAsyncClientHttpRequestInterceptor asyncTracingClientHttpRequestInterceptor() {
+      return (TracingAsyncClientHttpRequestInterceptor)
+          TracingAsyncClientHttpRequestInterceptor.create();
+    }
+  }
+
+  @Configuration
+  protected static class TraceInterceptorConfiguration {
+
+    @Autowired(required = false)
+    @SuppressWarnings("initialization.fields.uninitialized")
+    private Collection<org.springframework.web.client.AsyncRestTemplate> restTemplates;
+
+    @Autowired
+    @SuppressWarnings("initialization.fields.uninitialized")
+    private TracingAsyncClientHttpRequestInterceptor clientInterceptor;
+
+    @PostConstruct
+    public void init() {
+      if (this.restTemplates != null) {
+        for (org.springframework.web.client.AsyncRestTemplate restTemplate : this.restTemplates) {
+          List<org.springframework.http.client.AsyncClientHttpRequestInterceptor> interceptors =
+              new ArrayList<org.springframework.http.client.AsyncClientHttpRequestInterceptor>(
+                  restTemplate.getInterceptors());
+          interceptors.add(this.clientInterceptor);
+          restTemplate.setInterceptors(interceptors);
```

Comment:
Is there any concern that you may overwrite existing interceptors in the restTemplate?

Reply:
I am adding it to existing interceptor.